### ISSUE TITLE
already initialized constant Net::ProtocRetryError warnings

### ISSUE THREAD

**Issue by jvendetti** [ncbo/bioportal_web_ui#239]
After the upgrades to Rails 6.1 and Ruby 2.7, there's a handful of (so far benign) warnings that appear when the server starts:

```
/Users/vendetti/.rbenv/versions/2.7.7/lib/ruby/2.7.0/net/protocol.rb:66: warning: already initialized constant Net::ProtocRetryError
/Users/vendetti/.rbenv/versions/2.7.7/lib/ruby/gems/2.7.0/gems/net-protocol-0.2.1/lib/net/protocol.rb:68: warning: previous definition of ProtocRetryError was here
/Users/vendetti/.rbenv/versions/2.7.7/lib/ruby/2.7.0/net/protocol.rb:206: warning: already initialized constant Net::BufferedIO::BUFSIZE
/Users/vendetti/.rbenv/versions/2.7.7/lib/ruby/gems/2.7.0/gems/net-protocol-0.2.1/lib/net/protocol.rb:214: warning: previous definition of BUFSIZE was here
/Users/vendetti/.rbenv/versions/2.7.7/lib/ruby/2.7.0/net/protocol.rb:503: warning: already initialized constant Net::NetPrivate::Socket
/Users/vendetti/.rbenv/versions/2.7.7/lib/ruby/gems/2.7.0/gems/net-protocol-0.2.1/lib/net/protocol.rb:541: warning: previous definition of Socket was here
```

**Comment by syphax-bouazzouni** [comment]
the solution was to add the following to Gemfile
```ruby
gem "net-ftp", "~> 0.2.0", require: false
gem "net-http"
```

**Comment by jvendetti** [comment]
@syphax-bouazzouni - was there a particular reason you included the net-ftp gem in your comment? Adding net-http was sufficient to silence the warnings on my end.

**Comment by syphax-bouazzouni** [comment]
> @syphax-bouazzouni - was there a particular reason you included the net-ftp gem in your comment? Adding net-http was sufficient to silence the warnings on my end.

Hi @jvendetti , for `net-ftp`  I don't think it was directly related to this to this issue, and sincerely I don't remember really way.

But I think that we `require 'net/ftp'` in application_controller.rb and it didn't resolve

**Comment by alexskr** [comment]
deployment to stage/prod are failing when `gem net-http` is added:

```log
     01 bundle exec rake assets:precompile
      01 bundler: failed to load command: rake (/srv/ontoportal/bioportal_web_ui/shared/bundle/ruby/2.7.0/bin/rake)
      01 /usr/local/rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/bundler-2.3.12/lib/bundler/runtime.rb:309:in `check_for_activated_spec!': You have already activated uri 0.10.0, but your Gemfile requires uri 0.12.0. Since uri is a default gem, you can either remove your dependency on it or try updating to a newer version of bundler that supports uri as a default gem. (Gem::LoadError)
      01 	from /usr/local/rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/bundler-2.3.12/lib/bundler/runtime.rb:25:in `block in setup'
      01 	from /usr/local/rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/bundler-2.3.12/lib/bundler/spec_set.rb:136:in `each'
      01 	from /usr/local/rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/bundler-2.3.12/lib/bundler/spec_set.rb:136:in `each'
      01 bundler: failed to load command: rake (/srv/ontoportal/bioportal_web_ui/shared/bundle/ruby/2.7.0/bin/rake)
      01 /usr/local/rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/bundler-2.3.12/lib/bundler/runtime.rb:309:in `check_for_activated_spec!': You have already activated uri 0.10.0, but your Gemfile requires uri 0.12.0. Since uri is a default gem, you can either remove your dependency on it or try updating to a newer version of bundler that supports uri as a default gem. (Gem::LoadError)
```

running `bundle exec rails asset:precompile` fails in staging/production but seem to work in my local mac env.
updating bundler to latest version and 'gem update --system` doesn't seem to solve the issue

**Comment by jvendetti** [comment]
Alternative commands for asset generation that appear to work in the staging environment:

```ruby
RAILS_ENV=staging bin/rails assets:precompile
RAILS_ENV=staging bin/rake assets:precompile
```

**Comment by alexskr** [comment]
we use capistrano/rails for deployments and it uses 'bundle exec rake' to run migration/asset operations

**Comment by alexskr** [comment]
updating the bundler to v2.4 and gem system to 3.4.5 and running `bundle clean --force` seems to solve this issue

I propose to roll back this change and focus on making code base compatible with ruby v3 #251

**Comment by jvendetti** [comment]
This is fixed as of the [6.13.0](https://github.com/ncbo/bioportal_web_ui/releases/tag/v6.13.0) release
